tsup.config.base.ts

```diff
@@ -91,13 +91,16 @@ if (process.env.NODE_ENV === 'production') {
 interface ConfigOptions {
   name: string
   entry: string
+  buildFilter?: Array<BuildOptions["name"]>
 }
 
 export const defaultConfig = (
-  { name: prefix, entry }: ConfigOptions,
+  { name: prefix, entry, buildFilter }: ConfigOptions,
   options: Options
 ): Options[] => {
-  const artifactOptions: Options[] = buildTargets.map(
+  const artifactOptions: Options[] = buildTargets
+    .filter(target => !buildFilter || buildFilter.includes(target.name))
```

Comment:
@joshuaellis about removing the ESM build for React Native.

 I use the filter here but I have an issue. I am new to turbo, so I surely miss something.

Here is a view of my logs during the build. In pink, the log for the native target show that the build are well filtered.
![image](https://github.com/user-attachments/assets/7c031203-3043-4580-9f9c-22c8796efce3)

But the ESM flavor files still lands in the dist folder.

![image](https://github.com/user-attachments/assets/b980098b-d72f-4471-a9bd-02adc3831a98)

Have you any idea why?

Reply:
Sorry why do we want to remove ESM from react-native?

Is that what other libraries do? That feels weird.

Reply:
I explain it here https://github.com/pmndrs/react-spring/pull/2368#issuecomment-2824536087

Reply:
@joshuaellis I explain this here https://github.com/pmndrs/react-spring/pull/2368#issuecomment-2824536087

Reply:
Solved [here](https://github.com/pmndrs/react-spring/pull/2368/files#diff-b53466a39825cc473ab3e1acaea45dddf86aa08fe69a3ab8bd38c3193dfcc361R150) @joshuaellis 
The esOptions being present, it seems to automatically trigger the ES build regardless of the module resolution context. So I put a condition inside.

Test still passes, no regression detected.

Reply:
I have already build a few pre release versions of my Sproutch UI library including online demos:
- [RN example](exp+://expo-development-client/?url=https://u.expo.dev/29b4794d-fe0b-48d8-8705-26b7751d96cd/group/c3510727-a9b2-4c55-a666-8ff753329ddf)
- [Web example](https://dagatsoin.github.io/sproutch/?path=/docs/introduction--docs)
My stack: expo + storybook + RN web.